src/mahoji/lib/abstracted_commands/duelCommand.ts

```diff
@@ -151,8 +167,19 @@ export async function duelCommand(
 		if (selection.customID === 'CONFIRM') {
 			return await confirm(amount);
 		}
-	} catch (err) {
-		return cancel();
+	} catch (err: any) {
+		const code = typeof err === 'object' && 'code' in err ? err.code : null;
+		if (code === 'INTERACTION_COLLECTOR_ERROR') {
+			return cancel();
+		}
+		let failError = err;
+		if (!(failError instanceof Error)) {
+			failError = new Error(`Duel failed for unknown reason: ${String(err)}`);
+			failError.name = 'duelCommand Error';
+		}
+		return duelFail(failError);
 	}
-	return cancel();
+	const failError = new Error("duelCommand fell through try/catch this shouldn't happen");
+	failError.name = 'duelCommand Error';
+	return duelFail(failError);
```

Comment:
this seems really unnecessary. worst case just put it in a finally{}

Reply:
We don't want a finally because that will run every time and won't tell us anything.

Because the original code has `return cancel()` i can't rule out that that's where the error came from. (There is no guaranteed return in the try{} block.